services/discourse_translator/base.rb

```diff
@@ -18,32 +18,56 @@ def self.cache_key
       "#{key_prefix}#{access_token_key}"
     end
 
-    def self.translate(post)
+    def self.translate(object, target_language = I18n.locale)
       raise "Not Implemented"
     end
 
-    def self.detect(post)
+    def self.detect(object)
       raise "Not Implemented"
     end
 
     def self.access_token
       raise "Not Implemented"
     end
 
-    def self.from_custom_fields(post)
-      post_translated_custom_field = post.custom_fields[DiscourseTranslator::TRANSLATED_CUSTOM_FIELD] || {}
-      translated_text = post_translated_custom_field[I18n.locale]
+    def self.from_custom_fields(object, language)
+      translated_custom_field = object.custom_fields[DiscourseTranslator::TRANSLATED_CUSTOM_FIELD] || {}
+      translated_text = translated_custom_field[language]
 
       if translated_text.nil?
         translated_text = yield
 
-        post.custom_fields[DiscourseTranslator::TRANSLATED_CUSTOM_FIELD] =
-          post_translated_custom_field.merge(I18n.locale => translated_text)
+        object.custom_fields[DiscourseTranslator::TRANSLATED_CUSTOM_FIELD] =
+          translated_custom_field.merge("#{language}" => translated_text)
 
-        post.save!
+        object.save!
       end
 
       translated_text
     end
+
+    def self.get_text(object, max_length = nil)
+      case object.class.name
+      when "Post"
+        text = object.cooked
+        text = text.truncate(max_length, omission: nil) if max_length
+        text
+      when "Topic"
+        object.title
+      else
+        nil
+      end
+    end
+
+    def self.get_custom_field(object)
+      case object.class.name
+      when "Post"
+        DiscourseTranslator::DETECTED_LANG_CUSTOM_FIELD
+      when "Topic"
+        DiscourseTranslator::DETECTED_TITLE_LANG_CUSTOM_FIELD
+      else
+        nil
+      end
+    end
```

Comment:
Kind of related to https://github.com/discourse/discourse-translator/pull/79/files#r956832812, I'll prefer if we can implement different methods for dealing with `Post` and `Topic` so that we can avoid all these conditionals trying to figure out what type of object we're dealing with.